File: week_8_and_9/2_JavaScript/translate_to_js/my_solution.js
// U3.W8-9: Challenge you're converting


// I worked on this challenge [by myself, with: ].

// 2. Pseudocode



// 3. Initial Solution

function super_fizzbuzz(array) {
    var newArray = [];
    for (var i in array) {
        if ((array[i] % 3 === 0) && (array[i] % 5 === 0)) {
            newArray.push("FizzBuzz");
        } else if (array[i] % 3 === 0) {
            newArray.push("Fizz");
        } else if (array[i] % 5 === 0) {
            newArray.push("Buzz");
        } else {
            newArray.push(array[i]);
        }
    }
    return newArray;
}


// 4. Refactored Solution






// 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE






// 5. Reflection
/*
This was a fun challenge beacause I had just worked through it in ruby and I wanted to see
what it would look like in javascript. Not surprisingly, it looked pretty similar, as the
languages are not so far apart. On the whole I like ruby more than javascript, but I like
javascript a lot too, as where ruby feels fluid and versatile, javascript feels more
precise and mechanical.

File: week_8_and_9/3_Ruby/fizz_buzz/my_solution.rb
# U3.W8-9: 


# I worked on this challenge [by myself, with: ].

# 2. Pseudocode



# 3. Initial Solution

def super_fizzbuzz(array)
    new_array = []
    for i in 0...array.length
        if (array[i] % 3 == 0) && (array[i] % 5 == 0)
            new_array.push("FizzBuzz")
		elsif array[i] % 3 == 0
            new_array.push("Fizz")
        elsif array[i] % 5 == 0
            new_array.push("Buzz")
        else
            new_array.push(array[i])
        end
    end
    return new_array
end



# 4. Refactored Solution






# 1. DRIVER TESTS/ASSERT STATEMENTS GO BELOW THIS LINE






# 5. Reflection 
=begin
I found this test pretty simple, and did not take very long. I wrote it out more or less as-is,
only got confused when it came to test it. Ruby handles ranges a little bit different than I'm
used to so once I got that sorted out the rest was pretty simple.

